### PR TITLE
`Quiz exercises`: Fix an issue with the multiple choice visual mode

### LINKED ISSUE #9281: `Quiz:` the `visual` edition option does not save changes
### Describe the bug

The `visual` edition option does not save changes 

### To Reproduce

1. Create a MC question using the edit view
2. Save it 
3. Add options and explanations (optional)
4. Save it and verify that the question does not show the last changes 

### Expected behavior

Should save the changes 

### Screenshots

_No response_

### Which version of Artemis are you seeing the problem on?

7.5.1

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by pzdr7**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).

### Motivation and Context
Closes #9281.

When using the multiple choice visual mode, it was possible for the changes to be discarded when saving the quiz. This happened because the component retrieved the text from the wrong source (the outdated markdown editor, rather than the visual component)

### Description
When preparing to save while the editor is in visual mode, retrieves the text from the visual editor and stores it in the markdown editor. This triggers an update of the question and stores all changes.

Also, added a small e2e Test that checks if the Visual editor works as intended.

### Steps for testing

Prerequisites:
- 1 Instructor
- 1 Quiz exercise with a MC question

1. Navigate to the quiz and edit it.
2. Go to the MC question and switch to the Visual tab.
3. Modify the question. Make sure your changes are saved when you click save at the bottom (without having to switch to a different editor tab).

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
	- Enhanced editing experience for multiple-choice questions with improved markdown handling.
	- Introduced a visual mode for creating and editing multiple-choice questions.
	- Added unique identifiers for answer options and buttons to improve accessibility.

- **Bug Fixes**
	- Improved logic for saving markdown content based on editing mode.

- **Tests**
	- Expanded test coverage for multiple-choice question creation and editing in both visual and markdown modes.

### CODE REVIEW

File: src/test/playwright/support/pageobjects/exercises/quiz/QuizExerciseCreationPage.ts [hunk 1]
```diff
@@ -27,9 +28,34 @@ export class QuizExerciseCreationPage {
         const fileContent = await Fixtures.get('exercise/quiz/multiple_choice/question.txt');
         const textInputField = this.page.locator('.monaco-editor');
         await textInputField.click();
+        await clearTextField(textInputField);
```
Comment (pzdr7) [inline src/test/playwright/support/pageobjects/exercises/quiz/QuizExerciseCreationPage.ts:31]: Interestingly, this line missing was a mistake in the test case. The `fileContent` was getting appended to the default Artemis quiz template, resulting in a malformed quiz.

**Review by SimonEntholzer — APPROVED** [review]
Code

**Review by edkaya — APPROVED** [review]
Tested on TS4, works as expected

**Review by az108 — APPROVED** [review]
Code 👍

**Pull request merged**